### PR TITLE
Fix misfit config so it is possible to configure a subset of observations

### LINKED ISSUE #7211: Add config option to run misfit_preprocessor
Currently we run the misfit_preprocessor by the user having it as a hooked workflow: https://github.com/equinor/ert/blob/9496957720b5cfc7d2924f380d8d92c7d89cb0de/src/ert/cli/model_factory.py#L44-L49, which is not great, also they have lost the possibility to only allow a subset of observations. We should instead allow them to run it through:

`ANALYSIS_SET_VAR STD_ENKF MISFIT_PREPROCESS <OPTIONAL_OBSERVATIONS>`

where `<OPTIONAL_OBSERVATIONS` is a comma separated list of observations to scale, where we allow `*`-expansion. If the users provide nothing, we process all the observations.

We must also then give an error if the use the MISFIT_PREPROCESSOR workflow, and output what they should do instead.

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #7211

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/analysis/_es_update.py [hunk 1]
```diff
@@ -388,9 +391,12 @@ def _load_observations_and_responses(
     obs_mask = np.logical_and(ens_mean_mask, ens_std_mask)
 
     if misfit_process:
-        scaling[obs_mask] *= misfit_preprocessor.main(
-            S[obs_mask], scaled_errors[obs_mask]
-        )
+        for input_group in misfit_process:
+            group = _expand_wildcards(obs_keys, input_group)
```
Comment (oyvindeide) [inline src/ert/analysis/_es_update.py:395]: Not sure we want to do the expansion here, or if we want to do it earlier, a bit conflicted.

File: tests/unit_tests/config/test_analysis_config.py [hunk 2]
```diff
@@ -310,3 +310,18 @@ def test_incorrect_variable_deprecation_warning(config, expected):
...
     assert expected in [str(warning.message) for warning in all_warnings]
+
+
+@pytest.mark.parametrize(
+    "config",
+    [
+        ["OBSERVATIONS", "AUTO_SCALE", "OBS_*"],
+    ],
+)
+def test_misfit_configuration(config):
```
Comment (oyvindeide) [inline tests/unit_tests/config/test_analysis_config.py:321]: Will add more cases here.

File: src/ert/config/analysis_config.py [hunk 3]
```diff
@@ -247,10 +266,7 @@ def __eq__(self, other: object) -> bool:
         if self.stop_long_running != other.stop_long_running:
             return False
 
-        if self.std_cutoff != other.std_cutoff:
-            return False
-
-        if self.enkf_alpha != other.enkf_alpha:
+        if self.observation_settings != other.observation_settings:
```
[the remark this answers is not in the document]
Comment (sondreso) [reply src/ert/config/analysis_config.py:269]:
```python
>>> a = [["a"],["b"]]
>>> b = [["b"],["a"]]
>>> a == b
False
```

[the remark this answers is not in the document]
Comment (oyvindeide) [reply src/ert/config/analysis_config.py:271]: I would argue it does work, this is the behavior I would expect, as
```python
a = [["a"],["b"]]
b = [["a"],["b"]]
a == b
Out[1]: True
```

File: src/ert/analysis/_es_update.py (hunk 1 above)
Comment (oyvindeide) [reply src/ert/analysis/_es_update.py:296]: Cant agree with myself, in a potential future we might want to do the expansion when we start the base run model, and not when parsing the configuration, given that we will be input a list of observations to update. So I am actually leaning towards keeping it like this for now. But not sure. If we dont we need to do some context validation after we have finished parsing the config (this is parsed before the observations, so we dont know them yet).

File: src/ert/analysis/_es_update.py [hunk 4]
```diff
@@ -350,14 +344,23 @@ def _get_obs_and_measure_data(
...
+
 def _load_observations_and_responses(
     source_fs: EnsembleReader,
     alpha: float,
     std_cutoff: float,
     global_std_scaling: float,
     iens_ative_index: npt.NDArray[np.int_],
     selected_observations: List[Observation],
-    misfit_process: bool,
+    misfit_process: Optional[List[List[str]]],
```
[the remark this answers is not in the document]
Comment (oyvindeide) [reply src/ert/analysis/_es_update.py]: Added a suggestion:
```python
List[ObservationGroups]
```

File: src/ert/config/analysis_config.py [hunk 5]
```diff
@@ -64,6 +73,16 @@ def __init__(
         all_errors = []
 
         for module_name, var_name, value in analysis_set_var:
+            if module_name == "OBSERVATIONS":
+                if var_name == "AUTO_SCALE":
+                    observation_settings["auto_scale"].append(value.split(","))
+                else:
+                    all_errors.append(
+                        ConfigValidationError(
+                            f"Unknown module: {module_name} with variable: {var_name}"
```
[the remark this answers is not in the document]
Comment (oyvindeide) [reply src/ert/config/analysis_config.py]: Improved wording and added test

File: src/ert/config/analysis_config.py [hunk 6]
```diff
@@ -265,3 +291,14 @@ def __eq__(self, other: object) -> bool:
...
+
+ObservationGroups = List[str]
+
+
+@dataclass
+class UpdateSettings:
+    std_cutoff: float = 1e-6
+    alpha: float = 3.0
+    auto_scale_observations: List[ObservationGroups] = field(default_factory=list)
+    min_required_realizations: int = 2
```
Comment (sondreso) [inline src/ert/config/analysis_config.py:304]: Should we move these definitions further up in the file?

File: src/ert/config/analysis_config.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (oyvindeide) [reply src/ert/config/analysis_config.py:279]: Yes, it affects the result, as we can technically scale the same observation multiple times. That was always the case, but will influence the result. For example: ["OBS_1*,OBS*"] will potentially produce different results than: ["OBS*,OBS_1*"]

[the remark this answers is not in the document]
Comment (sondreso) [reply src/ert/config/analysis_config.py:279]: Alright, then I agree on the current behavior!

**Review by sondreso — APPROVED** [review]
LGTM!

**Pull request merged**